astropy/cosmology/_src/funcs/optimize.py

```diff
@@ -34,7 +32,7 @@ class _CustomSolverCallable(Protocol):
 
     def __call__(
         self, fun: Callable[..., Any], args: tuple[Any, ...], **kwargs: Any
-    ) -> OptimizeResult: ...
```

Comment:
Can this be changed to `"scipy.optimize.OptimizeResult"`
(and the `TYPE_CHECKING` changed to `import scipy.optimize`).
This allows for runtime type checkers to resolve the type annotation.